### PR TITLE
Split docs to kernel-mgmt

### PR DESCRIPTION
**Pull request by kevin-bates**
This pull request moves the kernel provider-specific material to jupyter-kernel-mgmt.

### CODE REVIEW

**Comment by kevin-bates** [comment]
This PR should be considered preliminary since it establishes the initial presence on readthedocs.io.  It also cross-references with jupyter_kernel_mgmt and https://github.com/takluyver/jupyter_kernel_mgmt/pull/26.

**Comment by takluyver** [comment]
I'm somewhat uncomfortable about splitting docs up still more - I think it's quite hostile to people trying to get to grips with Jupyter that documentation is in so many little separate pieces. But both JKM and jupyter_protocol are meant to be public API packages which should be documented, and that's easiest to do with the docs in their respective repos. And if jupyter_protocol becomes the main Python implementation of the message spec, it probably makes sense for the protocol description to live along with it.

So I think merging this and the corresponding JKM pull request is the way forwards. But I want to mention this reservation first, in case anyone can see a better way we could try to organise documentation.

**Comment by kevin-bates** [comment]
I agree with your concern.  We will also have **two** *"authoritative description[s] of the Jupyter messaging protocol"*. 😞 So, like many things between the `notebook` and `juptyer_server` repos, we'll need to keep things in-sync between `juptyer_client` and `jupyter_protocol/jupyter_kernel_mgmt`.  https://github.com/jupyter/jupyter_client/pull/486 is a great example, for instance.

File: docs/changelog.rst [hunk 1]
```diff
@@ -1,155 +1,14 @@
...
-- add :meth:`KernelManager.blocking_client`
-- provisional implementation of ``comm_info`` requests from upcoming 5.1 release of the protocol
+- Initial experimental implementation.
 
-4.0
-===
 
-The first release of Jupyter Client as its own package.
+Note: Because the code in this repository originated from jupyter_client you may 
+also want to look at its `release and changelog history <https://github.com/jupyter/jupyter_client/milestones/4.1>`__
```
Comment (takluyver) [inline docs/changelog.rst]: I guess this should point to https://jupyter-client.readthedocs.io/en/latest/changelog.html ?

File: docs/index.rst [hunk 2]
```diff
@@ -1,16 +1,17 @@
-Jupyter Client |version|
-========================
+Jupyter Protocol |version|
+==========================
 
-This package provides the Python API for starting, managing and communicating
-with Jupyter kernels.
+This package provides the Python API for message composition relative to Jupyter
```
Comment (takluyver) [inline docs/index.rst]: relative -> relevant?

File: docs/index.rst [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
+kernels. For information related to the launch, communication and lifecycle
+management of Jupyter kernels, please refer to the `Jupyter Kernel Management
+documentation <https://jupyter-kernel-mgmt.readthedocs.io/en/latest/index.html>`_.
 
 .. important::
    This document contains the authoritative description of the
    Jupyter messaging protocol. All developers are strongly encouraged to
```
Comment (takluyver) [inline docs/index.rst:11]: Let's comment this out or something until the focus has moved from jupyter_client. Maybe even add a note referring to the equivalent document in jupyter_client for now.

**Review by takluyver — COMMENTED** [review]
Thanks. We don't seem to have had any brainwaves for great alternatives, though. So let's move forwards with this.

File: docs/changelog.rst (hunk 1 above)
Comment (kevin-bates) [reply docs/changelog.rst]: Ok.  The docs are significantly out of date, but that's a different issue.  I agree that that's the correct reference.

**Pull request merged**